src/nvim/popupmenu.c

```diff
@@ -1395,17 +1405,33 @@ static void pum_position_at_mouse(int min_width)
 /// Select the pum entry at the mouse position.
 static void pum_select_mouse_pos(void)
 {
-  if (mouse_grid == pum_grid.handle) {
-    pum_selected = mouse_row;
+  int grid = mouse_grid;
+  int row = mouse_row;
+  int col = mouse_col;
+  win_T *wp = NULL;
+
+  if (grid == 0) {
+    wp = mouse_find_win(&grid, &row, &col);
+    // mouse_find_win compensates for offsets (e.g. 'winbar'), but we don't want that here.
+    if (wp) {
+      row += wp->w_winrow_off;
+      col += wp->w_wincol_off;
```

Comment:
ok we're doing this in 2 places in this very PR, so clearly we need `mouse_find_win_noadjust` :)